### ISSUE TITLE
Confusing .env documentation for SDK 49

### ISSUE THREAD

**Issue by jackymayo** [expo/eas-cli#1982]
### Do you understand that any feature requests opened in this repository will be closed?

Yes

### Version
Using SDK 49

## eas build loads from .env even if the environmental variables are defined in eas.json

From the [documentation
](https://docs.expo.dev/build-reference/variables/) 

> Environment variables are applied in the following order:
> 
> 1. eas.json build profile env field
> 2. Environment variables defined in EAS Secrets
> 3. .env files committed to source control and are not in .easignore

```json
    "build": {
        "production": {
            "env": {
                "EXPO_PUBLIC_BASE_URL": "https://www.ktbyte.com"
            }
        },
```

but my config was 

```bash
EXPO_PUBLIC_BASE_URL=http://10.0.0.76:4568
```

I had a headache trying to figure out why the login wasn't working and turns out it's pulling from .env in my preview build. 
So I had to rewrite my .env to this:
```bash
EXPO_PUBLIC_BASE_URL= https://www.ktbyte.com
```

My question is, what's the point of eas.json if eas build uses .env either way?
I specifically run `eas build --profile preview -p android --local`, is it a --local behavior?

**Comment by madhavgharmalkar** [comment]
Hey @jackymayo, I can confirm that I'm having the same issue.

Would like to see this get triaged as a bug or some documentation cleared up :)

**Comment by cbodonnell** [comment]
I have been running into similar issues with environment variables during builds.  Sharing my experience so far...

I have a default `.env` file defined that is committed to source control as documented [here](https://docs.expo.dev/guides/environment-variables/#using-multiple-env-files-to-define-separate-environments).  For building, I specify overrides in the `eas.json` build profile `env` field.  Based on what I see in the [docs](https://docs.expo.dev/build-reference/variables/#how-are-naming-collisions-between-secrets-the-env--field-in-easjson-and-env-files-handled), those defined in `eas.json` are loaded first, followed by the `.env` files, so it makes sense that my default `.env` file could still be taking priority.

In an attempt to get around this I tried creating a `.easignore` file with `.env` specified _and_ also tried leveraging the `EXPO_NO_DOTENV=1` variable in my build commands.  Neither of these worked and it still appeared to be loading variables from the default `.env` file for builds.  The only way so far that I have been able to get around this has been to explicitly exclude the default `.env` file from source control by adding it to `.gitignore`, which is not ideal.

Is it possible that I'm missing a step?

**Comment by jackymayo** [comment]
Oh sorry, I skimmed through the doc and realized just now it's the reverse behavior than what I was expecting.

**Comment by jackymayo** [comment]
@cbodonnell I think from the documentation here:
https://docs.expo.dev/build-reference/variables/#can-eas-build-use-env-files

## can-eas-build-use-env-files

> Environment variables defined in a .env file are only considered by the Expo CLI. Therefore, if you upload a .env file to EAS Build, it can be used to inline EXPO_PUBLIC_ variables into your application code.
> 
> However, the recommended practice is to use .env files in your local environment, while defining environment variables for EAS Build in eas.json. Environment variables defined in your eas.json build profile will be used when evaluating your app.config.js when running eas build and will be available to all steps of the build process on the EAS Build server.

Feel free to open your own issue but my original issue was misguided so I'm closing this.

**Comment by rlueder** [comment]
@cbodonnell in case you're still trying to figure this out, we recently moved from Buildkite to EAS and our Android builds somehow were all using the `.env.local` config.

Turns out we don't name our `.env` files [following the standard names](https://github.com/bkeepers/dotenv/blob/c6e583a/README.md#what-other-env-files-can-i-use) we use `.en.dev` and `.env.prod` instead of `.env.development` and `.env.production` so EAS was going straight to `.env.local` as it was the first valid match it'd find.

Solution was to run EAS commands setting `ENVFILE` first e.g. `ENVFILE=.env.prod npx eas build -p ios -e production`
